airflow/providers/google/cloud/hooks/dataplex.py

```diff
@@ -0,0 +1,247 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module contains Google Dataplex hook."""
+import os
+from time import sleep
+from typing import Any, Dict, Optional, Sequence, Union
+
+from google.api_core.retry import exponential_sleep_generator
+from googleapiclient.discovery import Resource, build
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+API_KEY = os.environ.get("GCP_API_KEY", "INVALID API KEY")
```

Comment:
Is that necessary?

Reply:
Yes, API Key is needed to perform operations on dataplex

Reply:
The authentication should be provided via dedicated connection. And as far as I remember `GoogleBaseHook` already provides all authentication methods supported by Google.

Reply:
Discussed within the team, for now I am going to remove API_KEY - it was needed for development purposes. Once the Dataplex API will be publicly available it will not be needed any more. I will commit changes and then draft this PR.